sdk/storage/azdatalake/internal/shared/challenge_policy.go

```diff
@@ -0,0 +1,112 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+package shared
+
+import (
+	"errors"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"net/http"
+	"strings"
+)
+
+type storageAuthorizer struct {
+	scopes   []string
+	tenantID string
+}
+
+func NewStorageChallengePolicy(cred azcore.TokenCredential, audience string) policy.Policy {
```

Comment:
Can you please also add support for `InsecureAllowCredentialWithHTTP`, see https://github.com/Azure/azure-sdk-for-go/pull/22712 for an example.

Reply:
Thanks!